tests/test_create.py

```diff
@@ -430,6 +432,29 @@ def test_zone_attr(self, provider, url):
             additional_args = " -H 'Metadata: true'"
         assert f"-Cnode.attr.zone=$(curl -s '{url}'{additional_args})" in cmd
 
+    def test_zone_attr_with_override(self):
+        with mock.patch(
+            "crate.operator.create.config.CLOUD_PROVIDER", CloudProvider.AZURE
+        ):
+            cmd = get_statefulset_crate_command(
+                namespace="some-namespace",
+                name="cluster1",
+                master_nodes=["node-0", "node-1", "node-2"],
+                total_nodes_count=3,
+                crate_node_name_prefix="node-",
+                cluster_name="my-cluster",
+                node_name="node",
+                node_spec={
+                    "resources": {"cpus": 1, "disk": {"count": 1}},
+                    "settings": {"node.attr.zone": "$(curl -s 'https://example.com')"},
```

Comment:
This should probably be moved to the `cluster_settings` to account for the issue pointed out above, where a cluster setting would still be overridden by the config option. And I'd probably use a simpler string for the zone value, e.g. `z1` or just "value" :man_shrugging:

Reply:
I figured it might be a good idea to have a separate test cases for node_spec and cluster_settings. Let me know if you disagree!

Reply:
Sounds good.